src/microscope_tiling_basler_camera/src/microscope_tiling_basler_camera/main.py

```diff
@@ -0,0 +1,152 @@
+OVERLAP_H: float = 0.5
+OVERLAP_V: float = 0.5
+
+# image capture
+SAVE_FOLDER: str = "tiles"
+RUN_BEST_EFFORT_STITCHING: bool = True
+RUN_NAIVE_TILING: bool = True
+
+# non user-controlled params
+CENTRE: NDArray[np.float64] = TOP_LEFT + (BOTTOM_RIGHT - TOP_LEFT) / 2.0
+EPSILON: float = 0.0000001
+
+
+def main() -> None:
+    """example entry point"""
+    check_user_specified_params()
+
+    if not os.path.exists(SAVE_FOLDER):
+        os.makedirs(SAVE_FOLDER)
+
+    with Connection.open_serial_port(SERIAL_PORT) as connection:
+        connection.detect_devices(identify_devices=True)
+        microscope: Microscope = Microscope.find(connection)
+        if microscope.plate is not None:
+            plate: AxisGroup = microscope.plate
+        else:
+            raise RuntimeError("Microscope plate is None")
+
+        print("Homing microscope plate")
+        plate.home()
+
+        camera: BaslerCameraWrapper = BaslerCameraWrapper()
+
+        path_builder: PathBuilder = PathBuilder(
+            PIXEL_WIDTH_MICRONS,
+            PIXEL_HEIGHT_MICRONS,
+            CAMERA_ROTATION_RAD,
+            camera.get_frame_width(),
+            camera.get_frame_height(),
+        )
+        tiling_path = path_builder.get_path_snake(
+            TOP_LEFT, BOTTOM_RIGHT, POINTS_UNITS, OVERLAP_H, OVERLAP_V
+        )
+
+        tiles = capture_images(tiling_path, camera, plate)
+        num_rows: int = len(tiling_path)
+
+        if RUN_BEST_EFFORT_STITCHING:
+            try:
+                try_stitch_images(tiles, num_rows)
+            except RuntimeError:
+                print("Stitching failed")
+        if RUN_NAIVE_TILING:
+            join_tiles(tiles, num_rows)
+
+
+def capture_images(
+    tiling_path: PathBuilder.Path,
+    camera: BaslerCameraWrapper,
+    plate: AxisGroup,
+) -> list[list[MatLike]]:
+    """
+        Move along provided path, capturing and saving an image at each point
+
+    Args:
+        tiling_path (PathBuilder.Path): the path generated from pathbuilder
+        camera (BaslerCameraWrapper): the camera which will be used to take pictures
+        plate (AxisGroup): the microscope plate
+
+    Raises:
+        RuntimeError: raised if camera API fails to capture an image for whatever reason
+
+    Returns:
+        list[list[MatLike]]: list of rows of captured images
+    """
+
+    tiles: list[list[MatLike]] = []
+    idx_y: int
+    grid_row: list[tuple[float, float]]
+    for idx_y, grid_row in enumerate(tiling_path):
+        tile_row: list[MatLike] = []
+        idx_x: int
+        point: tuple[float, float]
+        for idx_x, point in enumerate(grid_row):
+            plate.move_absolute(
+                Measurement(point[0], Units.LENGTH_MICROMETRES),
+                Measurement(point[1], Units.LENGTH_MICROMETRES),
+            )
+            try:
+                img = camera.grab_frame()
+            except ImageCaptureError as e:
+                print(e)
+                raise RuntimeError from e
```

Comment:
The exception will be printed anyway by the default handler.

Reply:
This will make it printed twice.

Reply:
Does this catch block do anything useful then?

Reply:
I meant to change this. Will update now

Reply:
fixed